spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/convert/DocumentAccessor.java

```diff
@@ -172,4 +172,21 @@ private static Document getOrCreateNestedDocument(String key, Bson source) {
 
 		return nested;
 	}
+
+	DocumentAccessor withCheckFieldMapping(boolean checkFieldMapping) {
+
+		if(!checkFieldMapping) {
+			return this;
+		}
+
+		return new DocumentAccessor(this.document) {
```

Comment:
If we want to handle field write checks here, how about creating a named subclass that we would use across `MappingMongoConverter` in the write paths?